src/providers/google/youtube-favourite.ts

```diff
@@ -27,143 +35,173 @@ export default class YouTubeFavourite extends GoogleHandler {
-            syncPosition.futureBreakId = serverResponse.data.items[0].id;
+            if (backfillResult.items.length) {
+                rangeTracker.completedRange({
+                    startId: backfillResult.items[0].sourceId,
+                    endId: nextPageToken
+                }, backfillResult.breakHit == SyncItemsBreak.ID);
+            } else {
+                rangeTracker.completedRange({
+                    startId: undefined,
+                    endId: undefined
+                }, backfillResult.breakHit == SyncItemsBreak.ID);
+            }
         }
 
-        if (_.has(serverResponse, "data.nextPageToken")) {
-            // Have more results, so set the next page ready for the next request
-            syncPosition.syncMessage = `Batch complete (${this.config.batchSize}). More results pending.`;
-            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        if (!items.length) {
+            syncPosition.syncMessage = `Stopping. No results found.`;
+            syncPosition.status = SyncHandlerStatus.ENABLED;
         } else {
-            syncPosition.syncMessage = "Stopping. No more results.";
-            syncPosition = this.stopSync(syncPosition);
+            if (items.length != this.config.batchSize && !nextPageToken) {
+                syncPosition.syncMessage = `Processed ${items.length} items. Stopping. No more results.`;
+                syncPosition.status = SyncHandlerStatus.ENABLED;
+            } else {
+                syncPosition.syncMessage = `Batch complete (${this.config.batchSize}). More results pending.`;
+            }
         }
 
-        return syncPosition;
+        syncPosition.thisRef = rangeTracker.export();
+
+        return {
+            results: items,
+            position: syncPosition,
+        };
     }
 
     protected async buildResults(
         serverResponse: GaxiosResponse<youtube_v3.Schema$VideoListResponse>,
         breakId: string,
         breakTimestamp?: string
-    ): Promise<SchemaFavourite[]> {
+    ): Promise<SyncFavouriteItemsResult> {
         const results: SchemaFavourite[] = [];
-    
-        const videos = serverResponse.data.items;
-    
-        for (const video of videos) {
+        let breakHit: SyncItemsBreak;
+
+        for (const video of serverResponse.data.items) {
             const videoId = video.id;
             const favouriteId = videoId;
-    
+
             if (favouriteId == breakId) {
                 const logEvent: SyncProviderLogEvent = {
                     level: SyncProviderLogLevel.DEBUG,
                     message: `Break ID hit (${breakId})`
-                }
-                this.emit('log', logEvent)
+                };
+                this.emit('log', logEvent);
+                breakHit = SyncItemsBreak.ID;
                 break;
             }
-    
+
             const snippet = video.snippet;
             const insertedAt = snippet.publishedAt || "Unknown";
```

Comment:
All date/time fields should be converted to a Date() object and then to `ISOString()` for extra guarantees that we get the correct date/time in the correct format.

Reply:
> All date/time fields should be converted to a Date() object and then to `ISOString()` for extra guarantees that we get the correct date/time in the correct format.

Done